firewall/privacy_mapper.go

```diff
@@ -812,6 +832,405 @@ func handleUpdatePolicyResponse(db firewalldb.PrivacyMapDB,
+			PendingForceClosingChannels: []*lnrpc.PendingChannelsResponse_ForceClosedChannel{},
+			WaitingCloseChannels:        []*lnrpc.PendingChannelsResponse_WaitingCloseChannel{},
+		}, nil
+	}
+}
+
+func handleClosedChannelsResponse(db firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	randIntn func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.ClosedChannelsResponse) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.ClosedChannelsResponse) (
+		proto.Message, error) {
+
+		closedChannels := make(
+			[]*lnrpc.ChannelCloseSummary,
+			len(r.Channels),
+		)
+
+		err := db.Update(func(tx firewalldb.PrivacyMapTx) error {
+			for i, c := range r.Channels {
+				var err error
+
+				remotePub := c.RemotePubkey
+				if !flags.Contains(session.ClearPubkeys) {
+					remotePub, err = firewalldb.HideString(
+						tx, remotePub,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				capacity, err := maybeHideAmount(
+					flags, randIntn, c.Capacity,
+				)
+				if err != nil {
+					return err
+				}
+
+				settledBalance, err := maybeHideAmount(
+					flags, randIntn, c.SettledBalance,
+				)
+				if err != nil {
+					return err
+				}
+
+				channelPoint := c.ChannelPoint
+				if !flags.Contains(session.ClearChanIDs) {
+					channelPoint, err = firewalldb.HideChanPointStr(
+						tx, c.ChannelPoint,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				chanID := c.ChanId
+				if !flags.Contains(session.ClearChanIDs) {
+					chanID, err = firewalldb.HideUint64(
+						tx, c.ChanId,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				closingTxid := c.ClosingTxHash
+				if !flags.Contains(session.ClearClosingTxIds) {
+					closingTxid, err = firewalldb.HideString(
+						tx, c.ClosingTxHash,
+					)
+					if err != nil {
+						return err
+					}
+				}
+
+				channel := lnrpc.ChannelCloseSummary{
+					// Non-obfuscated fields.
+					CloseInitiator: c.CloseInitiator,
+					CloseType:      c.CloseType,
+					OpenInitiator:  c.OpenInitiator,
+
+					// Obfuscated fields.
+					RemotePubkey:   remotePub,
+					Capacity:       capacity,
+					SettledBalance: settledBalance,
+					ChannelPoint:   channelPoint,
+					ChanId:         chanID,
+					ClosingTxHash:  closingTxid,
+				}
+
+				closedChannels[i] = &channel
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return &lnrpc.ClosedChannelsResponse{
+			Channels: closedChannels,
+		}, nil
+	}
+}
+
+func handleBatchOpenChannelRequest(db firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	_ func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.BatchOpenChannelRequest) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.BatchOpenChannelRequest) (
+		proto.Message, error) {
+
+		err := db.Update(func(tx firewalldb.PrivacyMapTx) error {
+			// Modify channels in place.
+			for _, c := range r.Channels {
```

Comment:
A general question here is whether we should take the override vs creation of a new struct approach. Could it be an advantage for forward compatibility if we don't have to manually add new fields that may be added to the api? Of course there's also a privacy tradeoff here as well.